src/pages/settings/Preferences/LanguagePage.js

```diff
@@ -47,14 +47,7 @@ const LanguagePage = (props) => {
             />
             <OptionsList
                 sections={[{data: localesToLanguages}]}
-                onSelectRow={
-                    (language) => {
-                        if (language.value !== props.preferredLocale) {
-                            App.setLocale(language.value);
-                        }
-                        Navigation.navigate(ROUTES.SETTINGS_PREFERENCES);
-                    }
-                }
+                onSelectRow={language => App.setLocale(language.value)}
```

Comment:
Now we no longer check if the value has changed before making the api call. Let's add that check within setLocale.

Reply:
yeah, I like the idea to add the check inside the action.